components/apimgt/org.wso2.carbon.apimgt.solace/src/main/java/org/wso2/carbon/apimgt/solace/utils/SolaceNotifierUtils.java

```diff
@@ -123,6 +125,59 @@ private boolean isSolaceGatewayDetailsAdded() {
         return true;
     }
 
+    /**
+     * Update Async API definition.
+     *
+     * @param api         API to be updated with async API definition in  Solace broker.
+     * @param environment Environment in which the API belongs to.
+     * @throws APIManagementException is error occurs when renaming the application.
+     */
+    public static void updateAsyncAPIDefinition(API api, Environment environment) throws
+            APIManagementException {
+        SolaceAdminApis solaceAdminApis = SolaceNotifierUtils.getSolaceAdminApis();
+        String organization = environment.getAdditionalProperties().get(SolaceConstants
+                .SOLACE_ENVIRONMENT_ORGANIZATION);
+        if (log.isDebugEnabled()) {
+            log.info("Updating Async API definition in Solace...");
+        }
+        String apiNameForRegistration = api.getId().getApiName() + "-" + api.getId().getVersion();
+        String apiDefinition = api.getAsyncApiDefinition();
+        Aai20Document aai20Document = (Aai20Document) Library.readDocumentFromJSONString(apiDefinition);
+        String[] apiContextParts = api.getContext().split("/");
+        String apiNameWithContext = environment.getName() + "-" + api.getId().getName() + "-" + apiContextParts[1] +
+                "-" + apiContextParts[2];
```

Comment:
Just for my understanding,  why do we need the context here.. 
Context is not only something like /abc/1.0.0. I can also be like /abc/qwe/1.0.0

Reply:
Yes you are right. Currently this feature supports only in super tenant domain. We will extend this feature to support other tenants later. 

This context name is is used to create the API product in Solace broker.